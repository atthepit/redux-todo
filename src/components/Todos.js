import React from 'react'
import Todo from './Todo'

const Todos = ({ onTodoClick, todos }) =>
  <ul>
    {todos.map(todo => <Todo key={ todo.id } onClick={ onTodoClick } todo={ todo } />)}
  </ul>

Todos.propTypes = {
  onTodoClick: React.PropTypes.func.isRequired,
  todos: React.PropTypes.array.isRequired,
}

export default Todos
