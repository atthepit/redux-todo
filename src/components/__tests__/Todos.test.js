import React from 'react'
import Todos from '../Todos'
import renderer from 'react-test-renderer'

describe(`Test Todos component`, () => {
  it(`renders correctly`, () =>{
    const onTodoClick = jest.fn()
    const todos = [
      { id: 0, text: `Hello world`, complete: false },
      { id: 1, text: `Hello!`, complete: true },
    ]
    const props = {
      onTodoClick,
      todos,
    }

    const tree = renderer.create(
      <Todos { ...props } />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
