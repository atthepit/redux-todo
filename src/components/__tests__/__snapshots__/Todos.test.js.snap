exports[`Test Todos component renders correctly 1`] = `
<ul>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }>
    Hello world
  </li>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "line-through",
      }
    }>
    Hello!
  </li>
</ul>
`;
